spec/controllers/v2/tailorings_controller_spec.rb

```diff
@@ -0,0 +1,137 @@
+          FactoryBot.create(
+            :v2_tailoring,
+            policy: parent,
+            profile: canonical_profiles.last,
+            os_minor_version: version
+          )
+        end.sort_by(&:id)
+      end
+
+      it_behaves_like 'collection', :policy
+      include_examples 'with metadata', :policy
+      it_behaves_like 'paginable', :policy
+      it_behaves_like 'searchable', :policy
+      it_behaves_like 'sortable', :policy
+    end
+  end
+
+  context '/policies/:id/tailorings/:id' do
+    describe 'GET show' do
+      let(:os_minor_version) { SecureRandom.random_number(10) }
+      let(:canonical_profile) do
+        FactoryBot.create(:v2_profile, ref_id_suffix: 'foo', supports_minors: [os_minor_version])
+      end
+      let(:parent) do
+        FactoryBot.create(
+          :v2_policy,
+          account: current_user.account,
+          profile: canonical_profile
+        )
+      end
+      let(:item) do
+        FactoryBot.create(
+          :v2_tailoring,
+          policy: parent,
+          profile: canonical_profile,
+          os_minor_version: os_minor_version
+        )
+      end
+
+      let(:extra_params) { { policy_id: parent.id, id: item.id } }
+      let(:notfound_params) do
+        extra_params.merge(policy_id: FactoryBot.create(
+          :v2_policy,
+          account: current_user.account,
+          profile: FactoryBot.create(:v2_profile)
+        ).id)
+      end
+
+      it_behaves_like 'individual', :policy
+    end
+  end
+
+  context '/policies/:id/tailorings/:id/tailoring_file' do
+    describe 'GET tailoring_file with randomly distributed rules' do
+      let(:canonical_profile) do
+        FactoryBot.create(
+          :v2_profile,
+          :with_rules,
+          security_guide: FactoryBot.create(:v2_security_guide, :with_rules, os_major_version: 9),
+          os_major_version: 9,
+          ref_id_suffix: 'bar',
+          supports_minors: [8]
+        )
+      end
+      let(:parent) do
+        FactoryBot.create(
+          :v2_policy,
+          account: current_user.account,
+          profile: canonical_profile
+        )
+      end
+      let(:extra_params) { { policy_id: parent.id, id: item.id } }
+      let(:item) do
+        FactoryBot.create(
+          :v2_tailoring,
+          policy: parent,
+          profile: canonical_profile,
+          os_minor_version: 8
+        )
+      end
+
+      it 'returns XCCDF tailoring file' do
+        header = JSON.parse(Base64.decode64(current_user.account.identity_header.raw))
+        header['identity']['auth_type'] = Insights::Api::Common::IdentityHeader::CERT_AUTH
+        request.headers['X-RH-IDENTITY'] = Base64.encode64(header.to_json)
+
+        get :tailoring_file,
+            params: extra_params.reject { |_, ep| ep.is_a?(ActiveRecord::Base) }.merge(parents: [:policy])
+        expect(response).to have_http_status :ok
+        expect(response.headers['Content-Type']).to eq('application/xml')
```

Comment:
Is the tailoring content actually tested? Also what about value definitions?

Reply:
Content is not tested. I've only verified it manually.

Reply:
Then what's the test for? And the factory around it?